### PR TITLE
SWATCH-587: use tally_instance_view to get records for instance api

### PR DESCRIPTION
**Pull request by kflahert**
https://issues.redhat.com/browse/SWATCH-587

Test Steps:

- Create test hosts: `bin/insert-mock-hosts   --num-guests=1   --num-hypervisors=1   --hypervisor-id=eb0d994a-e129-4a18-860d-3e39294f95df`
- Start app with: `DEV_MODE=true ./gradlew :bootRun`
- Run curl and should get records with VIRTUAL, PHYSICAL and HYPERVISOR measurement categories:
`curl -X 'GET'   'http://localhost:8000/api/rhsm-subscriptions/v1/instances/products/RHEL'   -H 'accept: application/json'   -H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6ImFjY291bnQxMjMiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsiaXNfb3JnX2FkbWluIjp0cnVlfSwiaW50ZXJuYWwiOnsib3JnX2lkIjoib3JnMTIzIn19fQo='`

### CODE REVIEW

**Comment by ntkathole** [comment]
Tests failing due to error in swatch api pod while calling instances api .

```
2023-01-12 07:34:46,921 [thread=http-nio-8000-exec-10] [ERROR] [org.candlepin.subscriptions.exception.mapper.BaseExceptionMapper] [Account: 7147355, OrgId: 14540822]- SUBSCRIPTIONS1000: An internal server error has occurred. Check the server logs for further details. - Cannot invoke "org.candlepin.subscriptions.db.model.TallyInstanceView.getKey()" because "tallyInstanceView" is null
java.lang.NullPointerException: Cannot invoke "org.candlepin.subscriptions.db.model.TallyInstanceView.getKey()" because "tallyInstanceView" is null
```

**Comment by kartikshahc** [comment]
@kflahert when I checked the json response it seems there is some duplication. Not sure if that is ok or not but, seems incorrect to me. Like the first two blocks have the same id, category, display name, mesurements, last seen. May not be something that this code have introduced.

Json response:
```
{
    "data": [
        {
            "id": "64ccda1a-c838-42a7-ab07-bd05bdb51bc8",
            "display_name": "1adff887-d987-41fa-8976-0443a1a94908",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        },
        {
            "id": "64ccda1a-c838-42a7-ab07-bd05bdb51bc8",
            "display_name": "1adff887-d987-41fa-8976-0443a1a94908",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        },
        {
            "id": "92eb31fc-34f6-4aa8-b2d8-38f2ee7933f3",
            "display_name": "48608929-2771-45b6-92a7-b1716262c635",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "number_of_guests": 8,
            "category": "HYPERVISOR"
        },
        {
            "id": "92eb31fc-34f6-4aa8-b2d8-38f2ee7933f3",
            "display_name": "48608929-2771-45b6-92a7-b1716262c635",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "number_of_guests": 8,
            "category": "HYPERVISOR"
        },
        {
            "id": "92eb31fc-34f6-4aa8-b2d8-38f2ee7933f3",
            "display_name": "48608929-2771-45b6-92a7-b1716262c635",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "number_of_guests": 8,
            "category": "PHYSICAL"
        },
        {
            "id": "92eb31fc-34f6-4aa8-b2d8-38f2ee7933f3",
            "display_name": "48608929-2771-45b6-92a7-b1716262c635",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "number_of_guests": 8,
            "category": "PHYSICAL"
        },
        {
            "id": "31aff1fa-92fb-4bb3-948f-21c4953c4875",
            "display_name": "c517db5a-1903-4c29-9922-5099eff9cf27",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        },
        {
            "id": "31aff1fa-92fb-4bb3-948f-21c4953c4875",
            "display_name": "c517db5a-1903-4c29-9922-5099eff9cf27",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        },
        {
            "id": "fc1626bc-6d21-4fd1-ae28-2747a46e434a",
            "display_name": "09329ec1-23ef-4c16-9532-76a2bb621331",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        },
        {
            "id": "fc1626bc-6d21-4fd1-ae28-2747a46e434a",
            "display_name": "09329ec1-23ef-4c16-9532-76a2bb621331",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        },
        {
            "id": "58c23409-4101-4d02-9afc-0468ca4cd484",
            "display_name": "42daf045-ee7a-4f0e-9a20-70dec265a4f9",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        },
        {
            "id": "58c23409-4101-4d02-9afc-0468ca4cd484",
            "display_name": "42daf045-ee7a-4f0e-9a20-70dec265a4f9",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        },
        {
            "id": "d67341e1-a1f2-4a92-abfe-0d85942931ff",
            "display_name": "6112172e-f806-4dee-a16b-6813b1941be8",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        },
        {
            "id": "d67341e1-a1f2-4a92-abfe-0d85942931ff",
            "display_name": "6112172e-f806-4dee-a16b-6813b1941be8",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        },
        {
            "id": "c8b444a9-2b8e-4763-8a3b-beff4a9ee797",
            "display_name": "20a413f0-2648-4bea-a5eb-35436e113cf7",
            "billing_provider": "_ANY",
            "billing_account_id": "_ANY",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "VIRTUAL"
        },
        {
            "id": "2cb2168e-3658-440a-9fd0-c5f083edc7fb",
            "display_name": "6e92ad1b-29be-4f4c-842e-1ff75ac9de9b",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        },
        {
            "id": "2cb2168e-3658-440a-9fd0-c5f083edc7fb",
            "display_name": "6e92ad1b-29be-4f4c-842e-1ff75ac9de9b",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        },
        {
            "id": "7a230ea1-ac6b-4834-84a4-726ae9b579e4",
            "display_name": "c487399f-6ce5-43ea-8609-c43647e25c36",
            "billing_provider": "_ANY",
            "billing_account_id": "_ANY",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "HYPERVISOR"
        },
        {
            "id": "7a230ea1-ac6b-4834-84a4-726ae9b579e4",
            "display_name": "c487399f-6ce5-43ea-8609-c43647e25c36",
            "billing_provider": "_ANY",
            "billing_account_id": "_ANY",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        },
        {
            "id": "cec516ea-f49b-4844-b181-1f98d07bf5ea",
            "display_name": "d233a8e9-72f8-4ab8-8160-d48c13c51309",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        },
        {
            "id": "cec516ea-f49b-4844-b181-1f98d07bf5ea",
            "display_name": "d233a8e9-72f8-4ab8-8160-d48c13c51309",
            "billing_provider": "",
            "measurements": [],
            "last_seen": "1993-03-26T00:00:00Z",
            "category": "PHYSICAL"
        }
    ],
    "meta": {
        "count": 21,
        "product": "RHEL",
        "measurements": []
    }
}
```

**Comment by kflahert** [comment]
@kartikshahc discussed this with David and this shouldn't be a problem because a UOM sort value is always passed when searching for rhel instances, which causes it to filter on UOM, so we won't get dupes. I also added UOM to the response so we can tell that there is a difference between records if it ever does get called without a UOM in the future.

File: src/main/java/org/candlepin/subscriptions/resource/InstancesResource.java [hunk 1]
```diff
@@ -203,32 +204,27 @@ protected void validateBeginningAndEndingDates(OffsetDateTime beginning, OffsetD
...
   private InstanceData asTallyHostViewApiInstance(
-      Host host, String monthId, List<String> measurements) {
+      TallyInstanceView tallyInstanceView, List<String> measurements) {
     var instance = new InstanceData();
     List<Double> measurementList = new ArrayList<>();
-    instance.setId(host.getInstanceId());
-    instance.setDisplayName(host.getDisplayName());
-    if (Objects.nonNull(host.getBillingProvider())) {
-      instance.setBillingProvider(host.getBillingProvider().asOpenApiEnum());
+    instance.setId(tallyInstanceView.getKey().getInstanceId().toString());
```
Comment (kartikshahc) [inline src/main/java/org/candlepin/subscriptions/resource/InstancesResource.java:210]: I assume this won't throw a NPE. Correct?

Comment (kflahert) [reply src/main/java/org/candlepin/subscriptions/resource/InstancesResource.java:210]: Since its part of they key it has to be non null so should be safe.

**Review by kartikshahc — APPROVED** [review]

**Pull request merged**